### PR TITLE
Add Support Win11 SnapLayout #1033

### PR DESCRIPTION
**Pull request by ghost1372**
Add Support Win11 SnapLayout #1033
Animation

### CODE REVIEW

File: src/Shared/HandyControl_Shared/Controls/Window/Window.cs [hunk 1]
```diff
@@ -361,23 +374,69 @@ private IntPtr HwndSourceHook(IntPtr hwnd, int msg, IntPtr wparam, IntPtr lparam
...
                     try
                     {
-                        _ = lparam.ToInt32();
+                        int x = lparam.ToInt32() & 0xffff;
+                        if (IsWin11SnaplayoutSupported())
+                        {
+                            int y = lparam.ToInt32() >> 16;
+                            var DPI_SCALE = DpiHelper.LogicalToDeviceUnitsScalingFactorX;
+                            var _button = WindowState == WindowState.Maximized ? _ButtonRestore : _ButtonMax;
+                            var rect = new Rect(_button.PointToScreen(
```
Comment (lindexi) [inline src/Shared/HandyControl_Shared/Controls/Window/Window.cs]: How about cache the Rect result? Calculation costs some resources and WM_NCHITTEST is called very frequently.

Comment (ghost1372) [reply src/Shared/HandyControl_Shared/Controls/Window/Window.cs]: @lindexi I think caching does not work properly,  I used the following code
```
if (rect.IsEmpty)
{
    rect = new Rect(_button.PointToScreen(
    new Point()),
    new Size(_button.Width * DPI_SCALE, _button.Height * DPI_SCALE));
}
```
But by resizing/moving window, the coordinates change and cached rect does not work and we need to call `rect = new Rect(_button.PointToScreen(` again.
Maybe I'm doing the caching operation wrong😕
Animation

**Comment by lindexi** [comment]
It seems be fixed by Windows Teams. See the issues in WPF repo.

**Comment by NaBian** [comment]
Is this branch done yet? @ghost1372

**Comment by ghost1372** [comment]
Hi @NaBian It is ready now😁

**Pull request closed without merging**